clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesHorizontalPodAutoscalerHandler.java

```diff
@@ -69,4 +78,86 @@ public Status status(KubernetesManifest manifest) {
+      Map<KubernetesManifest, List<KubernetesManifest>> relationshipMap) {
+    Map<String, KubernetesManifest> mapLabelToManifest = new HashMap<>();
+    Map<String, List<KubernetesManifest>> deploymentLabelsToReplicaSets = new HashMap<>();
+
+    allResources
+        .getOrDefault(KubernetesKind.REPLICA_SET, new ArrayList<>())
+        .forEach(
+            r -> {
+              addLabelsForKubernetesResource(mapLabelToManifest, r);
+              associateReplicaSetWithDeployments(deploymentLabelsToReplicaSets, r);
+            });
+
+    allResources
+        .getOrDefault(KubernetesKind.STATEFUL_SET, new ArrayList<>())
+        .forEach(
+            s -> {
+              addLabelsForKubernetesResource(mapLabelToManifest, s);
+            });
+
+    for (KubernetesManifest hpa :
+        allResources.getOrDefault(HORIZONTAL_POD_AUTOSCALER, new ArrayList<>())) {
+      relationshipMap.put(
+          hpa, getRelatedManifests(hpa, mapLabelToManifest, deploymentLabelsToReplicaSets));
+    }
+  }
+
+  private void addLabelsForKubernetesResource(
+      Map<String, KubernetesManifest> entries, KubernetesManifest resource) {
+    String resourceLabel =
+        resource.getNamespace() + " " + resource.getKindName() + " " + resource.getName();
+
+    entries.put(resourceLabel, resource);
+  }
+
+  private void associateReplicaSetWithDeployments(
+      Map<String, List<KubernetesManifest>> deploymentsToReplicaSets,
+      KubernetesManifest replicaSet) {
+    List<String> deploymentLabels =
+        replicaSet.getOwnerReferences().stream()
+            .filter(o -> o.getKind().toString() == KubernetesKind.DEPLOYMENT.toString())
+            .map(
+                d ->
+                    replicaSet.getNamespace()
+                        + " "
+                        + StringUtils.capitalize(d.getKind().toString())
+                        + " "
+                        + d.getName())
+            .collect(Collectors.toList());
+
+    for (String label : deploymentLabels) {
+      List<KubernetesManifest> replicaManifestsByDeployment = deploymentsToReplicaSets.get(label);
+      if (replicaManifestsByDeployment == null) {
+        replicaManifestsByDeployment = new ArrayList<>();
+      }
+      replicaManifestsByDeployment.add(replicaSet);
+      deploymentsToReplicaSets.put(label, replicaManifestsByDeployment);
+    }
+  }
+
+  private List<KubernetesManifest> getRelatedManifests(
+      KubernetesManifest hpa,
+      Map<String, KubernetesManifest> mapLabelToManifest,
+      Map<String, List<KubernetesManifest>> deploymentsToReplicaSets) {
+    Map<String, String> scaleTargetRef = hpa.getSpecScaleTargetRef().get();
+    String scaleTargetRefLabel =
+        hpa.getNamespace() + " " + scaleTargetRef.get("kind") + " " + scaleTargetRef.get("name");
+
+    KubernetesManifest matchingReplicaSet = mapLabelToManifest.get(scaleTargetRefLabel);
+
+    List<KubernetesManifest> result = new ArrayList<>();
```

Comment:
The UI expects autoscalers and load balancers to be tied to server groups. It will solely render HPA and load balancer badges on Replica Sets / Stateful Sets. A larger refactor of the pre-existing pattern on the front end could introduce these badges to ServerGroupManager components for autoscalers / load balancers. Given the extent of that change, I think we should utilize the current design for now.